content/pytorch/concepts/tensors/tensors.md

```diff
@@ -0,0 +1,229 @@
+
+The following demonstrates creating a matrix (2-dimensional tensor) with dimension sizes of 2 and 3:
+
+```python
+import torch
+
+myTensor = torch.empty(2,3)
+print(myTensor)
+```
+
+Output:
+
+```shell
+tensor([[0., 0., 0.],
+        [0., 0., 0.]])
+```
+
+> **Note:** The values contained in this tensor are not guaranteed and depend on the values already present at the relevant location in memory.
+
+### Data-Specific Tensors
+
+If the values within a tensor do matter, use the following methods:
+
+- For all zeros: `torch.zeros()`
+- For all ones: `torch.ones()`
+- For specified values: `torch.tensor()`
+- For all randomly-generated values: `torch.rand()`
+
+The following examples demonstrate creating data-specific tensors:
+
+```python
+import torch
+
+print(torch.zeros(3,2),"\n")
+print(torch.ones(1,4),"\n")
+print(torch.tensor([[1,0,1.56],[3.24, 0.5, 6]]),"\n")
+print(torch.rand(2,2,2))
+```
+
+Output:
+
+```shell
+tensor([[0., 0.],
+        [0., 0.],
+        [0., 0.]])
+
+tensor([[1., 1., 1., 1.]])
+
+tensor([[1.0000, 0.0000, 1.5600],
+        [3.2400, 0.5000, 6.0000]])
+
+tensor([[[0.2608, 0.5454],
+         [0.2085, 0.7901]],
+
+        [[0.8943, 0.7604],
+         [0.7584, 0.1678]]])
+```
+
+## Tensor Datatypes
+
+The datatype of a tensor determines the types of values that it can contain. By default, tensors contain 32-bit floats. However, tensors can also contain the following datatypes:
+
+- Floats (floating point numbers)
+  - 16-bit: `dtype=torch.float16`
+  - 32-bit: `dtype=torch.float32` (default)
+  - 64-bit: `dtype=torch.float64`
+- Complex numbers
+  - 32-bit: `dtype=torch.complex32`
+  - 64-bit: `dtype=torch.complex64`
+  - 128-bit: `dtype=torch.complex128`
+- Signed and unsigned integers
+  - 8-bit: `dtype=torch.int8` and `dtype=torch.uint8`
+  - 16-bit: `dtype=torch.int16` and `dtype=torch.uint16`
+  - 32-bit: `dtype=torch.int32` and `dtype=torch.uint32`
+  - 64-bit: `dtype=torch.int64` and `dtype=torch.uint64`
+- Quantized integers
+  - 8-bit: `dtype=torch.qint8`
+  - 32-bit: `dtype=torch.qint32`
+- Booleans: `dtype=torch.bool`
+
+### Declaring and Altering Tensor Datatypes
+
+Tensor datatypes can be declared at the creation of a datatype, and can be changed with the method `.to()`:
+
+```python
+import torch
+
+#Create a boolean tensor by explicit declaration
+myBoolTensor = torch.tensor([[1, 0],[0,0]], dtype=torch.bool)
+
+#Create a float16 tensor by explicit declaration
+myExFloat16Tensor = torch.tensor([[3.4, 0.9], [2.3, 1.3]], dtype=torch.float16)
+
+#Convert the float16 tensor to an int16 tensor using .to()
+myExFloat16Tensor = myExFloat16Tensor.to(torch.int16)
+
+print(myBoolTensor,"\n")
+print(myExFloat16Tensor)
+```
+
+Output:
+
+```shell
+tensor([[ True, False],
+        [False, False]]) 
+
+tensor([[3, 0],
+        [2, 1]], dtype=torch.int16)
+```
+
+Notice that when printing a tensor with a numerical datatype other than the default `torch.float32`, the datatype will be specified in the output.
```

Comment:
This is also not a note. It's not a qualification of something that may be misleading, it's simply drawing attention to a feature exhibited by the example. (Just because the word "notice" is similar to "note" does not mean they are synonymous.)